# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
#data['children'].value_counts()
#заметил, что есть отрицательные значения и вызывает сомнение цифра 20, но все возможно))
data['children'] = data['children'].abs()
#data['children'].value_counts()
#отрицательные значения измененя на положительные
#mean = data.groupby('children')['debt'].mean()
#def children_count(series):
    #return "(0:.2%)".format(series.sum()/series.count())
#data.groupby('children').agg({'children':['mean', 'count', children_count]})
#data.groupby('children')['debt'].mean()
ch = data.pivot_table(index = ['children'], values='debt', aggfunc = ('mean', lambda X:
                                                                       X.count()))

def percent(x):
    return "{0:.2%}".format(x)

ch['mean'] = ch['mean'].apply(percent)
ch = ch.rename(columns={"<lambda_0>": "Количество записей", "mean": "Среднее"})
ch.sort_values(by='Количество записей', ascending=False)

Количество записей Среднее
children                            
0                    14091.0   7.54%
1                     4808.0   9.23%
2                     2052.0   9.45%
3                      330.0   8.18%
4                       41.0   9.76%
5                        9.0   0.00%

**Вывод:** Люди, у которых есть от 1 до 4 детей, имеют более высокий процент долга, чем люди, у которых нет детей. Для людей, у которых 5 человек; Без объяснений.

конечно «разбег» между "цифрами" в наших данных мал. **Возможно**, что в рамках имеющихся данных сделать вывод о том, что семьи с детьми относительно семей без оных имеют какие-то задержки в выплатах кредитов нельзя. Слишком уж близки доли, чуть измениться выборка и значения могут измениться в другую сторону. То есть можно сделать вывод, что сколь угодно значимой разницы между группами нет ... однако учти, что даже небольшие разницы могут существенно сказываться на кредитном портфеле банка. Например, наш банк в год выдает кредитов на 100.000.000.000 (сто миллиардов) рублей. В какую сумму выльется разница в 1,9%, если мы выдадим кредит не той аудитории? Значимая разница в деньгах?

Однако, если считать, что **любая** разница между группами есть повод для «разделения» классов, то я бы сделал такой вывод: кредитополучатели без детей являются более ответственными, чем клиенты, у которых есть дети. Видимо, наличие ребенка накладывает на клиента дополнительные финансовое бремя (обязательства) и клиенты либо менее ответственно относятся к своим кредитным обязательствам, либо не имеют возможности в срок оплачивать обязательства.

Ну и что ещё можно увидеть в наших данных (агрегация «по count» в этом может помочь), то, что по некоторым группам заемщиков у нас недостаточно данных. Говорят, что выборка не сбалансирована, классы в выборке имеют разные размеры.

P.S. И вот ещё какой важный момент – да, на больших суммах и маленькие доли процента выльются в значительные объемы. Но вообще, насколько я понимаю логику кредитного бизнеса, дело не в том, чтобы кому-то не давать денег, а в том, чтобы компенсировать риск выдачи кредитным процентом. То есть, потенциальный должник заплатит более высокий процент за использование кредита. Так что в этой задаче фактически рассматривается вопрос - повышать ли процент заемщику с детьми. И статистика, которая показывает, что человек с 4 детьми просрочит выплату примерно так же, как и с двумя, - она, на мой взгляд, не обосновывает ужесточение кредитных условий для семей с детьми. К тому же тут зафиксировано количество детей только на момент выдачи кредита, никто не гарантирует, что их не станет больше в течение срока выплаты. Таким образом из этих данных можно сделать скорее другой вывод: кредитный портфель бездетных (со сроками выданных кредитов по несколько лет) скорее всего со временем ухудшится по показателю возврата денег (какая-то часть бездетных заведут детей, и у них должников из 7,5% станет 9%). И тогда этот риск можно было бы закладывать заранее (но для этого хорошо было бы понимать, сколько бездетных перейдут в категорию «с детьми» в течение 5 лет, например).

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [23]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
#print(data.groupby('family_status')['debt'].mean())
fs = data.pivot_table(index = ['family_status'], values='debt', aggfunc = ('mean', lambda X:
                                                                       X.count()))

def percent(x):
    return "{0:.2%}".format(x)

fs['mean'] = fs['mean'].apply(percent)
fs = fs.rename(columns={"<lambda_0>": "Количество записей", "mean": "Среднее"})
fs.sort_values(by='Количество записей', ascending=False)

Количество записей Среднее
family_status                                    
женат / замужем                   12261.0   7.56%
гражданский брак                   4134.0   9.31%
Не женат / не замужем              2796.0   9.76%
в разводе                          1189.0   7.06%
вдовец / вдова                      951.0   6.62%

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [24]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
inc = data.pivot_table(index = 'total_income_category', values='debt', aggfunc = ('mean', lambda X:
                                                                       X.count()))
def percent(x):
    return "{0:.2%}".format(x)

inc['mean'] = inc['mean'].apply(percent)
inc = inc.rename(columns={"<lambda_0>": "Количество записей", "mean": "Среднее по задолжностям"})
inc.sort_values(by='Количество записей', ascending=False)

Количество записей Среднее по задолжностям
total_income_category                                            
C                                 15921.0                   8.50%
B                                  5014.0                   7.06%
D                                   349.0                   6.02%
A                                    25.0                   8.00%
E                                    22.0                   9.09%

**Вывод:** люди с общей доходной категорией C имеют высокий процент долга, как и люди с совокупной доходной категорией A.
это может быть потому, что большинство людей относятся к категории С; долги могут стать высокими из-за многих потребностей, о которых вам нужно позаботиться.
для людей категории А дорогие товары и товары могут увеличить их долги

Быстрый взгляд на данные (в случае графика - это конечно ещё более наглядным становиться): ожидаемо, для группы с низким доходом (Е) процент невозвратов самый высокий. Что более интересно, так это то, что с ростом доходов тенденция (D->C и B->А) к снижению числа долгов не сохраняется, а скорее даже растёт. Можно сказать, что по какой-то причине люди "перестают считать деньги" или планировать "наперёд". Из-за подобного поведения наших групп параметр ("доход") нельзя назвать надёжным при определении платёжеспособности клиента. Без дополнительной информации он может даже наоборот привести к выдаче кредита состоятельному, но не слишком "надёжному" клиенту, хотя, возможно, это может быть аномалия, вызванная недостаточным размером выборки для клиентов с категорией доходов А.    
    
А теперь мой вывод- дополнительный, альтернативный вариант твоему: исходя из здравого смысла логично предположить, что чаще всего должниками становятся люди с небольшим доходом, которым финансовые проблемы не позволяют вовремя осуществлять платежи. Но глядя на наши результаты лично мой вывод таков: выборка по заемщикам, как с наименьшим доходом, так и с наибольшим крайне недостаточна для формирования однозначных выводов. Можно думать, что E-категория самая низкооплачиваемая не может корректно распоряжаться деньгами и «влезает» в кредиты, но тогда неясно почему у А-категории также достаточно высокий процент задолженности. В общем вывод по группам А и Е неочевиден, слишком мало данных.

Если же взять для анализа только группы В, С и D, то по полученным данным не наблюдаем зависимости "те, кто получает больше ЗП, лучше возвращает кредит». В целом это логично, так как материальный достаток - не единственная переменная, которая влияет на качество кредитной истории.

Но вот если пойти ещё дальше, отбросить малочисленные группы, оставив только две самые большие выборки (более-менее сопоставимые по размеру) B и C, то тогда зависимость можно сформулировать так: что заемщики с более высоким уровнем дохода реже допускают просрочку.

p.s. Однако 2174 пропуска в данных столбца "total_income" ставят под сомнение полученный нами результат. Сводная таблица (или группировка) с данными о зависимости между уровнем дохода и возвратом кредита в срок возможно могла бы выглядеть иначе, если бы не пропущенные данные, которые мы выше заполнили медианным значением зарплаты, исходя из типа занятости соответствующего заемщика.


#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [25]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
ps = data.pivot_table(index = ['purpose_category'], values='debt', aggfunc = ('mean', lambda X:
                                                                       X.count()))
#ps = data.pivot_table(index = ['purpose_category'], values='debt', aggfunc = ('mean', lambda X:
                                                                       #X.count()))
def percent(x):
    return "{0:.2%}".format(x)

ps['mean'] = ps['mean'].apply(percent)
ps = ps.rename(columns={"<lambda_0>": "Количество записей", "mean": "Среднее по задолжностям"})
ps.sort_values(by='Количество записей', ascending=False)

Количество записей Среднее по задолжностям
purpose_category                                                    
операции с недвижимостью             10751.0                   7.26%
операции с автомобилем                4279.0                   9.35%
получение образования                 3988.0                   9.25%
проведение свадьбы                    2313.0                   7.91%

<div class  = "alert alert-success">

**Вывод:** похоже, что у людей, принадлежащих ко второй и третьей строкам таблицы «purpose_category», самый высокий возврат в срок в процентах.
Это может показать нам, что люди, которые готовятся к свадьбе или пытаются расстаться с недвижимостью, более ответственны.
<div>

Всё подтверждается и цифрами, и здравым смыслом. Машины люди покупают в кредит гораздо дорогие, нежели могут соответствовать им своим доходом (ну кому не хочется красивую иномарку). А вот рисковать ипотечным жильём люди не хотят. Интересно конечно, что с образованием не так - как правило такие кредиты берут люди мотивированные, имеющие цель впереди, да и суммы таких кредитов судя по рынку онлайн образования не так велики. Ну и получается, что ещё более мотивированные люди берут деньги на организацию своей свадьбы)))

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* возможные причины появления пропусков в исходных данных связаны **с заменой значений пропусков** *нульями* или *заполняя их медиану*

In [26]:
data.isna().sum()

children                 0
days_employed            0
dob_years                0
education                0
education_id             0
family_status            0
family_status_id         0
gender                   0
income_type              0
debt                     0
total_income             0
purpose                  0
total_income_category    0
purpose_category         0
dtype: int64

*Вменение* - означает замену **нулей/пропусков** средними значениями **"Медианами"**.
<div class  = "alert alert-success">

Благодаря вменению наши данные не имеют нулей. Это может помочь лучше читать данные (также в некоторых процедурах машинного обучения программа может не иметь возможности читать нули).
Вменение не изменит значения наших данных данных.
<div>

В наших данных пропуски в столбцах "трудовой стаж" и "ежемесячный доход" абсолютно идентичны. Количество строк с отсутствующими данными в обоих указанных столбцах совпадает и равно 2174. Маловероятно, что специалисты по выдаче кредитов, заполнив все без исключения ячейки других столбцов по каждому кредитополучателю, оставляли незаполненными именно два показателя, и сделали это ровно 2174 раза в каждом столбце. Даже не проводя проверку, во всех ли случаях незаполненные ячейки находятся в одной и то же строке с высокой долей вероятности можно утверждать, что произошла техническая ошибка при формировании таблицы с исходными данными (либо некий сбой во время выгрузки данных на анализ, либо при объединении данных в единый файл, который также должен был передаваться на анализ, либо эти данные не собирались). 

На пользу основной версии об «объединении», как причине проблем с данными работает вот ещё какой аргумент: пропуски в столбцах 'total_income' и 'days_employed' соответствуют значениям 'пенсионер', 'госслужащий', 'компаньон', 'сотрудник' и 'предприниматель' в столбце 'income_type'. Однако в 'income_type' также присутствуют значения ‘студент’, ‘безработный’ и ‘в декрете’, в которых пропусков нет вообще. На основании этого можно предположить, что исходная таблица была объединена с другим датасетом, содержавшим сведения о категориях 'пенсионер', 'госслужащий', 'компаньон', 'сотрудник' и 'предприниматель', но без сведений о доходе и трудовом стаже. В результате при объединении таблиц столбцы 'total_income' и 'days_employed' были заполнены значением NaN.

Это моё ИМХО (врачи говорят – «второе мнение»).



#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* лучшее решение заполнить пропуски медианным значением потому что, В лучшем случае эти информационные пробелы:
1. Может не позволить нам прийти к ключевым выводам или…
2. Может привести к тому, что любые выводы, полученные на основе данных, будут иметь сомнительную точность или…
3. Может нарисовать ограниченную картину мира (представьте, что вы пытаетесь описать комнату, глядя только в замочную скважину).
4. … и в худшем случае они являются препятствиями, ведущими к плохим моделям машинного обучения — мусор на входе, мусор на выходе.

### Шаг 4: общий вывод.

Есть зависимости между целями кредита, семейным положением, уровнем дохода и количеством детей в семье. Нельзя сказать, что они кардинально отличаются, но 2% на больших числах могут оказать существенное влияние на кредитный портфель. Особое внимание следует уделить вдовам/вдовцам или молодоженам — с ними могут риски по невозврату кредитов выше.
Надо отметить на то, что ответственность клиента влияет тоже на кредиты.


#### **Oбщий вывод:**

Было проведено исследование, чтобы сформулировать вывод о причинах, которые могут повлиять на увеличение долга.
данные для изучения - статистика о платежеспособности клиентов.
Результаты будут учтены при постреонии модели крдитного скоринга - специальной системы, которая оценивает способность потенциального заемщика вернуть кредит банку.

*Перед исследования были поставлены несколько гипотиз:*

**Гипотиза** 1. Колтчество детей вдияет на факт погашения возврата крдита

**Гипотиза** 2. Семейные положения вдияет на факт погашения возврата крдита.

**Гипотиза** 3. Уровень дохода вдияет на факт погашения возврата крдита. 

**Гипотиза** 4. Цель кредита вдияет на факт погашения возврата крдита. 

В результате удалось выявить зависимости:

**Гипотиза** 1: *частично подверждена* нужны больше данных по клиентам с 3, 4, и 5 детей для более точных выводов.

- 7.54% доля просрочных кредитов у заёмщиков без детей.
- 9.23% доля просрочных кредитов у заёмщиков с 1 ребенком
- 9.45% доля просрочных кредитов у заёмщиков с 2 детей

у заёмщиков без детей более ответвенные чем тех заёмщиков, у кого есть детей (по данным с 1 и 2 детей).

**Гипотиза** 2: *подверждена* 

- у не женатых клиентов (9.76%) и тех, у кого официально не зарегистрирован брак (9.31%).Выше вероятность стать должником.

- те, кто в разводе (7.06%), женат/замужем (7.56%) или вдовец/вдова (6.62%) менее вероятно становятся должниками.

**Гипотиза** 3: *частично подверждена* "для более точных выводов необходимо больше данных по клиентам с уровнем дохода категории D (от 30 001 до 50 000) А(от 1 000 001), Е( до 30 000)

- (7.06%) - доля просроченных кредитов у заёмщиков с уровнем дохода от 200.001 до 1.000.000;
- (8.5%) - доля просроченных кредитов у заёмщиков с уровнем дохода от 50.001 до 200.000;

**Гипотеза** 4 *подтверждена*:

- 7.26% доля просроченных кредитов, целью которых были операции с недвижимостью;
- 9.35% доля просроченных кредитов, целью которых было проведение свадьбы;
- 9.25% доля просроченных кредитов, целью которых были операции с автомобилем;
- 7.91% доля просроченных кредитов, целью которых было получение образования.

**Общие Рекомендации**:

1. Специалистам, которые занимаются построением модели кредитного скоринга, можно использовать зависимости из второй и четвёртой гипотезы (семейное попозжение и цели кредита)
2. Необходимо собрать более полную выборку для повторной проверки первой и третьей гипотез

